apps/server/lib/lexical/server/provider/queue.ex

```diff
@@ -136,7 +136,28 @@ defmodule Lexical.Server.Provider.Queue do
       Queue.Supervisor.run_in_task(handler)
     end
 
+    alias Lexical.Protocol.Types.ErrorCodes
+    require ErrorCodes
+
     defp write_reply(response) do
+      case Convert.to_lsp(response) do
+        {:ok, lsp_response} ->
+          Transport.write(lsp_response)
+
+        error ->
+          Logger.critical(
+            "Failed to convert into a response #{inspect(error)}  #{inspect(response)}"
+          )
+
+          response_error =
+            ResponseError.new(
+              code: :internal_error,
+              message: "Failed to convert #{response.__struct__} due to #{inspect(error)}"
+            )
+
+          Transport.write(%{id: response.id, error: response_error})
+      end
+
```

Comment:
@dimitarvp @scohen 

After spending some time, I found out that these lines of code were causing `nvim` to crash repeatedly. This code responds more than twice to the same `id`, which we need to avoid.

![CleanShot 2024-04-03 at 21 19 12@2x](https://github.com/lexical-lsp/lexical/assets/12830256/9d71bdd3-7bad-429a-a37a-3d1551678754)

When I tried checking out this file from the main branch, everything worked fine

```bash
git checkout main apps/server/lib/lexical/server/provider/queue.ex
```

Reply:
@dimitarvp can you look in the lexical logs for the error that's causing the double write?

Reply:
@scohen There are several hundred log lines that seem relevant (at least they both contain `elixir-ls` and `lexical`), do you want them all?

Reply:
I have zero such lines in my log though. Hmmm. I'll try to do another experiment today or tomorrow and see if I can make it manifest itself (and will also post any findings about trying to list workspace symbols).

Reply:
@dimitarvp i've pushed a fix to this branch that removes the double-write, so you should be able to use lexical now. 

just let it run and whenever you have a chance, grep your logs for "Failed to convert" and open an issue.

Reply:
Thanks, @scottming for debugging this.